NDTensors/src/lib/Sectors/src/category_definitions/su.jl

```diff
@@ -58,26 +75,100 @@ function Base.show(io::IO, ::MIME"text/plain", s::SU)
     i += 1
   end
 
-  println(io, "└─", "┴─"^max(0, l[i] - 1), "┘")
+  print(io, "└─", "┴─"^max(0, l[i] - 1), "┘")
   return nothing
 end
 
 #
 # Specializations for the case SU{2}
-# Where irreps specified by dimension "d"
 #
 
-dimension(s::SU{2}) = 1 + label(s)[1]
+# optimize implementation
+quantum_dimension(s::SU{2}) = category_label(s)[1] + 1
 
-SU{2}(d::Integer) = SU{2}((d - 1, 0))
+GradedAxes.dual(s::SU{2}) = s
 
-dual(s::SU{2}) = s
-
-function fusion_rule(s1::SU{2}, s2::SU{2})
-  d1, d2 = dimension(s1), dimension(s2)
-  return [SU{2}(d) for d in (abs(d1 - d2) + 1):2:(d1 + d2 - 1)]
+function label_fusion_rule(::Type{<:SU{2}}, s1, s2)
+  labels = collect((i,) for i in (abs(s1[1] - s2[1])):2:(s1[1] + s2[1]))
+  degen = ones(Int, length(labels))
+  return degen, labels
 end
 
+# define angular momentum-like interface using half-integers
+SU2(h::Number) = SU{2,1}((HalfIntegers.twice(HalfIntegers.HalfInteger(h)),))
```

Comment:
```suggestion
SU2(h::Number) = SU{2}((HalfIntegers.twice(HalfIntegers.HalfInteger(h)),))
```
Probably best to avoid using the constructor `SU{N,M}` since that is an internal detail.

Reply:
I remember discussing this design choice a lot but don't remember the rational behind the final decision we made, could you give a brief summary to remind me why we landed on this design?

Reply:
I guess the basic idea is that `SU{2}(...)` and `SU2(...)` use different conventions for the label.

Reply:
Yes, `SU{2}` follows `SU` conventions for generic SU(N), where the label corresponds to the Young tableau of the irrep. It allows a unified interface for N=2 and larger N. `SU2` is a function that takes a half-integer as input and allows for a SU(2) specific interface in terms of spins. It then casts to `SU{2}` with the corresponding Young Tableau, which is `(2*s,)`.

Reply:
Thanks, that sounds familiar to me as the design we settled on in our previous discussion about this and it seems like a simple and nice design.